### PR TITLE
Use unicorn instead of webrick in replicant's systemd command

### PR DESCRIPTION
**Pull request by eldebrim**
Since we upgraded redmine we can now use unicorn as the backend server. This will resolve [29558](https://support.osuosl.org/Ticket/Display.html?id=29558)

### CODE REVIEW

File: files/default/unicorn.rb [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  # the following is *required* for Rails + "preload_app true",
+  defined?(ActiveRecord::Base) &&
+    ActiveRecord::Base.establish_connection
+
+  # if preload_app is true, then you may also want to check and
+  # restart any other shared sockets/descriptors such as Memcached,
+  # and Redis.  TokyoCabinet file handles are safe to reuse
+  # between any number of forked children (assuming your kernel
+  # correctly implements pread()/pwrite() system calls)
+end
```
Comment (ramereth) [inline files/default/unicorn.rb]: Why is this file included?

File: recipes/app2.rb [hunk 2]
```diff
@@ -193,6 +193,7 @@
     environment 'RAILS_ENV' => 'production'
     working_directory '/home/replicant/redmine'
     pid_file '/home/replicant/pids/unicorn.pid'
-    exec_start '/home/replicant/.rvm/bin/rvm 2.3.0 exec  ./bin/rails server -e production -b 140.211.9.86 -p 8090'
+    exec_start '/home/replicant/.rvm/bin/rvm 2.3.0 do bundle exec unicorn '\
+'-l 8090 -c unicorn.rb -E deployment -D'
```
Comment (ramereth) [inline recipes/app2.rb]: This should still fit within the 120char limit. Please don't wrap this.

File: files/default/unicorn.rb (hunk 1 above)
Comment (eldebrim) [inline files/default/unicorn.rb]: I wasn't sure whether to put the unicorn configuration file in production using chef or upload it manually. After testing the systemd command without including the `-c unicorn.rb` option, I realized that the unicorn server would still be up without it. Should I still include unicorn.rb or remove the `-c unicorn.rb` option?

Comment (ramereth) [inline files/default/unicorn.rb]: IMO that's outside the scope of this cookbook. It should be manually copied there on the app side.

**Comment by ramereth** [comment]
@eldebrim please squash commits and also ensure the unicorn.rb file exists on the server before merging and deploying.

**Comment by eldebrim** [comment]
@ramereth @knightsamar OK. As for installing unicorn, should it be done by adding `gem "unicorn"` to `/home/replicant/redmine/Gemfile` then using the following command:
`[replicant@app2.osuosl.org redmine]$ .rvm/bin/rvm 2.3.0 do bundle install`
or is it handled by a configuration management tool?

**Comment by ramereth** [comment]
@eldebrim that is not currently handled via config management so we'd do it manually.

File: recipes/app2.rb [hunk 3]
```diff
@@ -193,6 +193,6 @@
     environment 'RAILS_ENV' => 'production'
     working_directory '/home/replicant/redmine'
     pid_file '/home/replicant/pids/unicorn.pid'
-    exec_start '/home/replicant/.rvm/bin/rvm 2.3.0 do bundle exec unicorn -l 8090 -c unicorn.rb -E production -D'
+    exec_start '/home/replicant/.rvm/bin/rvm 2.3.0 do bundle exec unicorn -l 8090 -c unicorn.rb -E production'
```
Comment (knightsamar) [inline recipes/app2.rb]: Quite interesting because iam-staging seems to use -D.

**Comment by eldebrim** [comment]
@knightsamar Yep. I was wrong

**Comment by eldebrim** [comment]
@knightsamar For the unicorn.rb file, the one iam-staging uses didn't work because it loads a scheduler script that isn't present in redmine. Is it okay if I use  [this one](unicorn.bogomips.org/examples/unicorn.conf.minimal.rb) instead?

**Comment by eldebrim** [comment]
!bump patch ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a patch-level version bump to v1.11.9.  Have a nice day!
